pkg/kubectl/deployment.go

```diff
@@ -56,18 +74,37 @@ func (s DeploymentBasicGeneratorV1) Generate(params map[string]interface{}) (run
 	if !isArray {
 		return nil, fmt.Errorf("expected []string, found :%v", imageStrings)
 	}
-	delegate := &DeploymentBasicGeneratorV1{Name: name, Images: imageStrings}
-	return delegate.StructuredGenerate()
+	return &BaseDeploymentGenerator{
+		Name:   name,
+		Images: imageStrings,
+	}, nil
 }
 
-// StructuredGenerate outputs a deployment object using the configured fields
-func (s *DeploymentBasicGeneratorV1) StructuredGenerate() (runtime.Object, error) {
-	if err := s.validate(); err != nil {
-		return nil, err
+// structuredGenerate: determine the fields of a deployment. The struct that
+// embeds BaseDeploymentGenerator should assemble these pieces into a
+// runtime.Object.
+func (b BaseDeploymentGenerator) structuredGenerate() (
+	podSpec v1.PodSpec,
+	labels map[string]string,
+	selector metav1.LabelSelector,
+	err error,
+) {
+	err = b.validate()
+	if err != nil {
+		return
 	}
+	podSpec = buildPodSpec(b.Images)
+	labels = map[string]string{}
+	labels["app"] = b.Name
+	selector = metav1.LabelSelector{MatchLabels: labels}
+	return
+}
 
+// buildPodSpec: parse the image strings and assemble them into the Containers
+// of a PodSpec. This is all you need to create the PodSpec for a deployment.
+func buildPodSpec(images []string) v1.PodSpec {
```

Comment:
fyi, I expect the args to this function to grow a lot since most of the flags driven values will be for things in the podspec.  Consider making a function of `BaseDeploymentGenerator` and access the fields directly.